Update branch

Dubkoslove opened the pull request.

maab committed:
Example suggestion for a visibility field

maab committed:
Update datasetIds.md

maab committed:
Update datasetIds.md

JohannesPeterseil committed:
Update titles.md

(cherry picked from commit 4b07c575d44df6c0db5ddbce014ae04d601d3fab)

JohannesPeterseil committed:
Update assetType.md

(cherry picked from commit 794d5144b7108eb13b825b9ca2d018130175a5a5)

JohannesPeterseil committed:
Update authors.md

(cherry picked from commit 30dc0b0726de85d468d8c5c0f6df9e9e3162c6c2)

JohannesPeterseil committed:
Update authors.md

(cherry picked from commit 33af9072d5d93bbcf9236e9244a4df9277e8a8be)

JohannesPeterseil committed:
Update authors.md

(cherry picked from commit b9e9aecb7812c069f19dd31ccfb0d82bd0cdb9b9)

maab committed:
Update assetType.md

Adds all eLTER Standard Observation (SO) Types to the asset type enumeration.

maab committed:
Update descriptions.md

Specified input validation rules and added placeholder/tootip text

maab committed:
Update geoLocations.md

Added placeholder/tooltip texts

maab committed:
Update licenses.md

Added placeholder and input validation requirements and some comments.

maab committed:
Update descriptions.md

Consolidated tooltip/placeholder field

maab committed:
Update geoLocations.md

Consolidated tooltip/placeholder field

maab committed:
Update licenses.md

Consolidated tooltip/placeholder field

Dubkoslove committed:
Remove pws.

Dubkoslove committed:
Add missing metadata files

maab committed:
Add temporal resoulution to temporal coverage

maab committed:
Add instrumentation section

maab committed:
Add SO Habitat definition

maab committed:
Add Data Level field description

maab committed:
Rename authors "Full Name" to Affiliation

Kulivox commented on docs/new_docs/elter_model/temportalCoverages.md:
```diff
@@ -25,6 +26,29 @@
       }
     }
+  },
+  "temporalResolution": {
```
@maab Shouldn't this be a file metadata attribute? Because I can imagine there are going to be many datasets that contain multiple files with different temporal resolutions.

maab replied:
It should indicate a general/overall temporal resolution of the dataset. If the dataset contains more then one file the lowest frequency should be reported.

maab commented on docs/new_docs/elter_model/temportalCoverages.md:
```diff
@@ -25,6 +26,29 @@
     }
+  },
+  "temporalResolution": {
+    "required": "once",
```
Latest discussions dicision is to make it optinal

maab committed:
Make temporal resoution optional

maab committed:
Affiliaition field for Contributors

maab committed:
Add DEIMS Location ID to Gegraphic Locations

JohannesPeterseil committed:
Create ResourceType

JohannesPeterseil committed:
Update ResourceType

JohannesPeterseil committed:
Update ResourceType

JohannesPeterseil committed:
Update ResourceType

JohannesPeterseil committed:
Create ResourceType.md

JohannesPeterseil committed:
Update ResourceType.md

JohannesPeterseil committed:
Delete docs/new_docs/elter_model/ResourceType

JohannesPeterseil committed:
Update ResourceType.md

JohannesPeterseil committed:
Update ResourceType.md

JohannesPeterseil committed:
Update ResourceType.md

JohannesPeterseil committed:
Update ResourceType.md

JohannesPeterseil committed:
Update ResourceType.md

JohannesPeterseil committed:
Update SOReference.md

JohannesPeterseil committed:
Update SOReference.md

JohannesPeterseil committed:
Update SOReference.md

JohannesPeterseil committed:
Update SOReference.md

JohannesPeterseil committed:
Update assetType.md

JohannesPeterseil committed:
Update assetType.md

JohannesPeterseil committed:
Update _base.md

JohannesPeterseil committed:
Update ecosystem.md

JohannesPeterseil committed:
Update dataLevel.md

JohannesPeterseil committed:
Update siteReference.md

JohannesPeterseil committed:
Update _template.md

JohannesPeterseil committed:
Update _template.md

JohannesPeterseil committed:
Update descriptions.md

JohannesPeterseil committed:
Update authors.md

JohannesPeterseil committed:
Update authors.md

JohannesPeterseil committed:
Update contributors.md

JohannesPeterseil committed:
Update keywords.md

JohannesPeterseil committed:
Update geoLocations.md

JohannesPeterseil committed:
Update publicationDate.md

JohannesPeterseil committed:
Update licenses.md

JohannesPeterseil committed:
Update temportalCoverages.md

JohannesPeterseil committed:
Update propertyRights.md

JohannesPeterseil committed:
Update propertyRights.md

JohannesPeterseil committed:
Update shortNames.md

JohannesPeterseil committed:
Create relatedIdentifiers.md

JohannesPeterseil committed:
Update relatedIdentifiers.md

JohannesPeterseil committed:
Create alternateIdentifiers.md

JohannesPeterseil committed:
Update _base.md

JohannesPeterseil committed:
Update project.md

Dubkoslove committed:
Merge remote-tracking branch 'origin/JohannesPeterseil-patch-1' into update_branch

# Conflicts:
#	docs/new_docs/elter_model/assetType.md
#	docs/new_docs/elter_model/authors.md
#	docs/new_docs/elter_model/contributors.md
#	docs/new_docs/elter_model/dataLevel.md
#	docs/new_docs/elter_model/descriptions.md
#	docs/new_docs/elter_model/geoLocations.md
#	docs/new_docs/elter_model/licenses.md

Dubkoslove commented on docs/new_docs/elter_model/ResourceType.md:
````diff
@@ -0,0 +1,77 @@
+## Schema
+```json
+{
+  "resource_type": {
````
As previously discussed, consolidating all resources under a single metadata schema and user interface is a poor approach. This setup is likely to lead to numerous issues and errors, some of which are inevitable.

A better solution is to organize the enum values into distinct groups, each with a (largely) consistent metadata schema. For each group, we can develop a dedicated model and user interface to better manage and streamline their handling.

Dubkoslove commented on docs/new_docs/elter_model/SOReference.md:
```diff
@@ -12,6 +12,9 @@
 {
   "SOReference": {
```
Note: This will be done via invenio vocabs in the future.

Dubkoslove commented on docs/new_docs/elter_model/alternateIdentifiers.md:
```diff
@@ -0,0 +1,45 @@
+          "type": "enum",
+          "label": "Related Identifier Type",
+          "tooltip": "The type of identifier.",
+          "enumValues": ["ARK", "arXiv", "bibcode", "DOI", "EAN13", "EISSN", "Handle", "ISBN", "ISSN", "ISTC", "LISSN", "LSID", "ORCID", "PMID", "PURL", "UPC", "URL", "URN", "w3id", "URI"]
```
Please maintain cohesive naming by using 'alternate' instead of 'related' for consistency.

Dubkoslove commented on docs/new_docs/elter_model/authors.md:
```diff
@@ -13,37 +13,77 @@
 {
   "creators": {
```
We moved organizations to the separated field.

Dubkoslove commented on docs/new_docs/elter_model/authors.md:
```diff
@@ -13,37 +13,77 @@
+                "label": "Scheme",
+                "tooltip": "The scheme used for the identifier.",
+                "default": "orcid",
+                "enum": [
```
I removed ror from the enum values. The ror was associated with the organization, not the person

Dubkoslove commented on docs/new_docs/elter_model/authors.md:
````diff
@@ -65,17 +105,17 @@ https://eml.ecoinformatics.org/schema/eml-resource_xsd.html#ResourceGroup_creato
 ```json
 {
-  "authors": [
+  "authors/creators": [
````
Decide which one

Dubkoslove replied:
Authors

Dubkoslove commented on docs/new_docs/elter_model/dataLevel.md:
````diff
@@ -11,13 +11,32 @@
 ```json
 {
   "dataLevel": {
````
values will be done via invenio vocabs in the future

Dubkoslove committed:
Update dataset identifier and URL formatting

Dubkoslove commented on docs/new_docs/elter_model/relatedIdentifiers.md:
````diff
@@ -0,0 +1,55 @@
+## Schema
+```json
+{
+  "relatedIdentifiers": {
````
What is the difference between alternateIds, relatedIds, and datasetIds? I want to ensure we have a clear definition for each.

What is the full use case for these fields? Have we documented (the use case) it somewhere? I'd like to analyze the available options and evaluate whether Invenio might provide a better solution for this scenario.

Dubkoslove commented on docs/new_docs/elter_model/shortNames.md:
```diff
@@ -1,5 +1,7 @@
 [Back to model](_base.md)
 
+**COMMENT:** Maybe not needed, as included in the titles (titles.md)
```
Agree :smile:

Dubkoslove commented on docs/new_docs/elter_model/titles.md:
```diff
@@ -14,14 +14,27 @@
+        "type": "enum",
+        "label": "Type",
+        "tooltip": "The type of title.",
+        "enum": enum [Title, Alternative Title, Subtitle, Translated Title, Other],
```
We’ve already discussed this. Do we want to complicate things, or should we store only a single title? Previously, you and Ulf suggested deleting all language variants (e.g., the list of title translations).

Dubkoslove committed:
Fix formatting

JohannesPeterseil committed:
Update relatedIdentifiers.md

JohannesPeterseil committed:
Update alternateIdentifiers.md

JohannesPeterseil committed:
Update alternateIdentifiers.md

JohannesPeterseil committed:
Update alternateIdentifiers.md

JohannesPeterseil committed:
Update alternateIdentifiers.md

JohannesPeterseil committed:
Update relatedIdentifiers.md

JohannesPeterseil committed:
Update titles.md

JohannesPeterseil committed:
Update authors.md

JohannesPeterseil committed:
Update contributors.md

JohannesPeterseil committed:
Update publicationDate.md

JohannesPeterseil committed:
Update language.md

JohannesPeterseil committed:
Update descriptions.md

JohannesPeterseil committed:
Update descriptions.md

JohannesPeterseil committed:
Update keywords.md

JohannesPeterseil committed:
Update licenses.md

JohannesPeterseil committed:
Update geoLocations.md

JohannesPeterseil committed:
Update temportalCoverages.md

JohannesPeterseil committed:
Update project.md

JohannesPeterseil committed:
Update project.md

Dubkoslove commented:
Decide whether to remove assetType / SOReference

maab committed:
Move temporalResolution from temporalCoverage to a separate section

maab committed:
Move temporalResolution from temporalCoverage to a separate section

JohannesPeterseil committed:
Update authors.md

JohannesPeterseil committed:
Update authors.md

JohannesPeterseil committed:
Update authors.md

JohannesPeterseil committed:
Update contributors.md

JohannesPeterseil committed:
Update contributors.md

JohannesPeterseil committed:
Update contributors.md

JohannesPeterseil committed:
Create responsibleOrganisations.md

JohannesPeterseil committed:
Update responsibleOrganisations.md

JohannesPeterseil committed:
Update responsibleOrganisations.md

JohannesPeterseil committed:
Update dataLevel.md

JohannesPeterseil committed:
Update alternateIdentifiers.md

JohannesPeterseil committed:
Update datasetIds.md

JohannesPeterseil committed:
Update titles.md

JohannesPeterseil committed:
Update titles.md

JohannesPeterseil committed:
Update shortNames.md

JohannesPeterseil committed:
Update habitatReference.md

JohannesPeterseil committed:
Update habitatReference.md

JohannesPeterseil committed:
Update habitatReference.md

JohannesPeterseil committed:
Update habitatReference.md

JohannesPeterseil committed:
Update instrumentation.md

JohannesPeterseil committed:
Update instrumentation.md

JohannesPeterseil committed:
Update instrumentation.md

JohannesPeterseil committed:
Update keywords.md

JohannesPeterseil committed:
Update propertyRights.md

JohannesPeterseil committed:
Update propertyRights.md

JohannesPeterseil committed:
Update project.md

Dubkoslove committed:
Remove files

JohannesPeterseil committed:
Update _base.md

JohannesPeterseil committed:
Update _base.md

JohannesPeterseil committed:
Update _base.md

JohannesPeterseil committed:
Update keywords.md

oggioniale committed:
Update the documentation for specific elements

Dubkoslove committed:
Rename dir model name - dir name

Dubkoslove committed:
Update actual complete schema

Dubkoslove committed:
Update assetType

Dubkoslove committed:
Update alternateIdentifiers.md

Dubkoslove committed:
Update relatedIdentifiers.md

Dubkoslove committed:
Remove shortnames, update titles

Dubkoslove committed:
Update authors -> creators

Dubkoslove committed:
Update contributors

Dubkoslove committed:
Update descriptions

Dubkoslove committed:
Update keywords

Dubkoslove committed:
Update licenses

Dubkoslove committed:
Update licenses

Dubkoslove committed:
Update geoLocations

Dubkoslove committed:
Update temporalCoverage

Dubkoslove committed:
Update taxonomicCoverages.md

Dubkoslove committed:
Update methods.md

Dubkoslove committed:
Update projects

Dubkoslove committed:
Update siteReferences

Dubkoslove committed:
Remove SOReference

Dubkoslove committed:
Remove ecosystem

Dubkoslove committed:
Remove files, geoServerInfo, Update dataLevel

Dubkoslove committed:
Remove resourceType

Dubkoslove committed:
Update temporal resolution

Dubkoslove committed:
Update habitatReference.md

Dubkoslove committed:
Removed instrumentation - methods and description can contain it...

Dubkoslove committed:
add Contact point to the schema

Dubkoslove committed:
Fix template

Dubkoslove committed:
Init external asset model

Dubkoslove committed:
Change taxonomic prefix values

maab committed:
Create eLTERMetadataSchema.json

Introduction of a single file JSON Schema representation of the eLTER Metadata Schema

maab committed:
Move eLTERMetadataSchema.json to schemas subdirectory

maab committed:
Adding authors section to eLTERMetadataSchema.json

Commit:
New authors and preparation to have multiple schema files

Commit:
Split schema into subschemas

maab committed:
Create eLTERMetadataSchemaExternalSources.json

maab committed:
Rename eLTERMetadataSchemaDarIngest.json to eLTERMetadataSchemaDatasets.json

maab committed:
Rename eLTERMetadataSchemaExternalSources.json to eLTERMetadataSchemaExternalDatasets.json

1O committed:
append blank line to each .json

to silence "incomplete last line" warning/error from AJV jsonschema validator

1O committed:
correct keywords

unknown keyword "$0id-comment" replaced with "$comment"
unknown keyword "authors" replaced with "x-authors" (`x-` indicating a custom annotation)

1O committed:
correct keywords

replace unknown keyword "_comment_on_$id" with "$comment"
replace unknown keyword "authors" with "x-authors", `x-` signalizing a custom annotation

1O committed:
add datasets schema as monolith

include all local subschemas into main schema

1O committed:
fix properties required from habitatReferences

replace `"required": true` with `"required": ["soHabitatCode", "soHabitatURI"]`

1O committed:
remove eLTERMetadataSchemaExternalDatasets.json from wrong directory

1O committed:
correct keywords and requirements notation

replace `_comment_on_$id` with `$comment`
replace `"required": true` with `"required": ["soHabitatCode", "soHabitatURI"]`

1O committed:
add monolithic schema for external datasets

pull in all referenced subschemas to obtain a monolithic schema without references to subschemas

1O committed:
add utilities for JSONschema construction

1O committed:
prefix custom keys with "x-"

1O committed:
unify $ids

following the pattern   "$id": "elter://metadata.schemas/subschemas/<filename>"
note: the id does not have to resolve to an URL on the internet; I chose this to prevent the validator from downloading stale versions hosted on GitHub, though.

1O committed:
Delete docs/new_docs/validate.R

didn't belong here

1O committed:
Create README.md

1O committed:
Update README.md

Dubkoslove committed:
Merge pull request #6 from eLTER-RI/elter-metadata-schema-as-json-schema

Create eLTERMetadataSchema.json
